aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -966,6 +966,22 @@ public <T extends S3ClientBuilder> void applyCredentialConfigurations(
             : awsClientProperties.credentialsProvider(accessKeyId, secretAccessKey, sessionToken));
   }
 
+  public <T extends S3AsyncClientBuilder> void applyCredentialConfigurations(
```

Comment:
Are these two different hierarchies? Just wondering why we need two versions of this but I'm not familiar with AWS's code here

Reply:
Great suggestion. There is a shared hierarchy targeted for `S3ClientBuilder` and `S3AsyncClientBuilder` that is part of the AWS public API, `S3BaseClientBuilder`.